Route name in RouteResult is route path not name

weierophinney opened the pull request:
Looking at the `RouteResult` in the `psrRequest` shows the route path such as `/customer/create` as the `matchedRouteName` rather than `customer.create` which is what is specified in the route configuration as such:

``` php
[
    'name' => 'customer.create',
    'path' => '/customer/create',
    'middleware' =>  App\Action\CreateCustomer::class,
    'allowed_methods' => ['POST'],
],
```

I'm not sure if this is the intention or not. If it isn't a bug like I suspect it is, what is the purpose of the name then?

The specific line is https://github.com/zendframework/zend-expressive-fastroute/blob/master/src/FastRouteRouter.php#L236

weierophinney commented:
I'll see if I can create a reproduce case, and get back to you.

weierophinney commented:
Verified; I'll get a fix in shortly.

nesl247 commented:
Awesome! Glad to see you get to this so quickly. Just wrote some code and came across this and it was holding me back, so this is great.

weierophinney committed:
Return route name, not path, in RouteResult

Updates `marshalMatchedRoute()` to return the matched route from its
internal `array_reduce()` operation, which ensures that we can marshal
`RouteResult` with the route *name* instead of the *path*.

weierophinney commented on test/FastRouteRouterTest.php:
```diff
@@ -154,7 +154,7 @@ public function testMatchingRouteShouldReturnSuccessfulRouteResult()
         $this->assertInstanceOf(RouteResult::class, $result);
         $this->assertTrue($result->isSuccess());
-        $this->assertEquals('/foo', $result->getMatchedRouteName());
+        $this->assertEquals('/foo^GET', $result->getMatchedRouteName());
```
A note to anybody who stumbles across this: by fixing the issue, I exposed a problem with this particular test. The assumption was that the route _name_ was simply the path. However, that's not the case, nor has it been, for most of the lifetime of this implementation; in order to distinguish between middleware using the same path but different HTTP methods, we had to append the method information auto-magically. Updating the code to return the name picked that up, and hence the change to this assertion.

weierophinney commented:
@nesl247 Fixed, and a new version, 1.0.1, is released with the fix; thanks for the report!
